addons/account/wizard/account_move_send.py

```diff
@@ -226,28 +226,21 @@ def _compute_checkbox_download(self):
         for wizard in self:
             wizard.checkbox_download = wizard.enable_download and wizard.company_id.invoice_is_print
 
-    @api.depends('move_ids')
+    @api.depends('mode', 'checkbox_send_mail')
     def _compute_send_mail_extra_fields(self):
         for wizard in self:
             wizard.enable_send_mail = wizard.mode in ('invoice_single', 'invoice_multi')
             wizard.display_mail_composer = wizard.mode == 'invoice_single'
-            send_mail_readonly = False
-
+            wizard.send_mail_readonly = not wizard.move_ids.filtered(lambda x: x.partner_id.email)
             display_messages = []
-            if wizard.enable_send_mail:
+            if wizard.mode == 'invoice_multi' and wizard.checkbox_send_mail:
                 invoices_without_mail_data = wizard.move_ids.filtered(lambda x: not x.partner_id.email)
```

Comment:
Thanks for the suggestion. It's much more clean now. I only changed the `send_mail_readonly` from your suggestion because the readonly should only be on if all the invoice don't have mail data, so that we can send partially send invoices like in the specs. Let me know what you think :)

```py
wizard.send_mail_readonly = invoices_without_mail_data == wizard.move_ids
```

Reply:
@yosuanicolaus I'm ok with that :)